lib/web/websocket/receiver.js

```diff
@@ -298,26 +298,30 @@ class ByteParser extends Writable {
       // defined as the status code (Section 7.4) contained in the first Close
       // control frame received by the application
       code = data.readUInt16BE(0)
+      if (!isValidStatusCode(code)) {
+        return null
+      }
+    } else {
+      return null
     }
 
     // https://datatracker.ietf.org/doc/html/rfc6455#section-7.1.6
-    /** @type {Buffer} */
-    let reason = data.subarray(2)
+    /** @type {number} */
+    let reasonOffset = 2
 
-    // Remove BOM
-    if (reason[0] === 0xEF && reason[1] === 0xBB && reason[2] === 0xBF) {
-      reason = reason.subarray(3)
+    // Detect BOM and ignore it
+    if (data.length >= 5 && data[2] === 0xEF && data[3] === 0xBB && data[4] === 0xBF) {
+      reasonOffset = 5
     }
 
-    if (code !== undefined && !isValidStatusCode(code)) {
-      return null
+    if (data.length - reasonOffset < 1) {
+      return { code, reason: '' }
     }
 
+    let reason = ''
```

Comment:
There is a misunderstanding here. Returning `null` means that the close frame's body is invalid. If we can't parse the reason, due to invalid utf-8, we **need** to return null here.

Reply:
This seems to be a false conclusion. We got a valid code, so we wont need to set the code 1005 or 1006. The message is somewhat corrupt. So. We need to prefer the valid code and discard the message.